packages/features/sales-escrows/modals/ConfirmSalesActionModal.tsx

```diff
@@ -135,17 +148,17 @@ export const ConfirmSalesActionModal = ({
         return handleClose(false);
       }
 
-      const escrowReceipt = {
-        seller: { principal: offer.seller.principal },
-        buyer: { principal: offer.buyer.principal },
+      const escrowReceipt: EscrowReceipt = {
+        seller: { principal: offer.seller['principal'] },
```

Comment:
You can you still use strong typing here by checking:
```
if ('principal' in offer.seller) {
  // write code with do notation like offer.seller.principal
}
```

Reply:
This type of logic is best done in a parser function so that it doesn't need to be repeated in all the components. The odcParser.ts file is a good place to add a new function for parsing the results of sale_nft_origyn and returning a simple data structure with everything the components need.

Reply:
However, given time constraints, we can refactor the code later.